index/scorch/segment/zap/intcoder.go

```diff
@@ -134,3 +139,42 @@ func (c *chunkedIntCoder) Write(w io.Writer) (int, error) {
 func (c *chunkedIntCoder) FinalSize() int {
 	return len(c.final)
 }
+
+// chunkLengthsToOffsets converts the chunk length array
+// to a chunk starting offset array. The readChunkBoundary
+// will figure out the start and end of every chunk from
+// these offsets. The starting offset of the first/single
+// array element will always be zero and this position is
+// used for storing the size of the current last item in
+// the array at any given point.
```

Comment:
i'm wondering instead of using the 0'th position to store the last item's info, how about logically shifting everything leftwards by one position?

That is, if you have a bunch of offsets for N item items, then actually only hold onto offsets[1:] of N-1 items?

Reply:
great point @steveyen, that is i believe the same situation and solution i had used for the "running" field in the index snapshot, which similarly remembers where certain global doc num ranges start.  there was no need to "remember" that it starts with 0, but there is a need to know how many are in the last segment, so it seems like the same case.

Reply:
thanks @steveyen , done!